src/main/java/de/komoot/photon/PhotonDoc.java

```diff
@@ -88,27 +93,50 @@ public PhotonDoc(PhotonDoc other) {
         this.context = other.context;
         this.country = other.country;
         this.state = other.state;
+        this.isExpanded = other.isExpanded;
     }
 
+    /**
+     * Get an id suitable for using as an ES document id
+     * 
+     * @return an id
+     */
+    @Nonnull
     public String getUid() {
-        if (houseNumber == null || houseNumber.isEmpty())
-            return String.valueOf(placeId);
-        else
-            return String.valueOf(placeId) + "." + houseNumber;
+        String id = getBaseId(osmType, osmId, tagKey);
+        if (houseNumbers == null || houseNumbers.isEmpty()) {
+            return id;
+        } else {
+            return id + (isExpanded ? "." + houseNumbers.get(0) : "");
+        }
```

Comment:
Always adding the housenumber looks wrong now with respect to the updates below. You don't add the housenumbers there (and can't because you don't have the old housenumber). Given that simple house number objects are not expanded anymore, you should only add the housenumber to interpolation documents. Then the updates work with using the simple base id.

Reply:
isExpanded is only set to true for interpolations.